lib/OpenQA/Git.pm

```diff
@@ -68,30 +70,68 @@ sub set_to_latest_master ($self, $args = undef) {
 sub commit ($self, $args = undef) {
     $self->_validate_attributes;
 
-    my @git = $self->_prepare_git_command($args);
     my @files;
 
     # stage changes
     for my $cmd (qw(add rm)) {
         next unless $args->{$cmd};
         push(@files, @{$args->{$cmd}});
-        my $res = run_cmd_with_log_return_error([@git, $cmd, @{$args->{$cmd}}]);
-        return _format_git_error($res, "Unable to $cmd via Git") unless $res->{status};
+        my $res = $self->_run_cmd([$cmd, @{$args->{$cmd}}]);
+        return $self->_format_git_error($res, "Unable to $cmd via Git") unless $res->{status};
     }
 
     # commit changes
     my $message = $args->{message};
     my $author = sprintf('--author=%s <%s>', $self->user->fullname, $self->user->email);
-    my $res = run_cmd_with_log_return_error([@git, 'commit', '-q', '-m', $message, $author, @files]);
-    return _format_git_error($res, 'Unable to commit via Git') unless $res->{status};
+    my $res = $self->_run_cmd(['commit', '-q', '-m', $message, $author, @files]);
+    return $self->_format_git_error($res, 'Unable to commit via Git') unless $res->{status};
 
     # push changes
     if (($self->config->{do_push} || '') eq 'yes') {
-        $res = run_cmd_with_log_return_error([@git, 'push']);
-        return _format_git_error($res, 'Unable to push Git commit') unless $res->{status};
+        $res = $self->_run_cmd(['push']);
+        return $self->_format_git_error($res, 'Unable to push Git commit') unless $res->{status};
     }
 
     return undef;
 }
 
+sub get_current_branch ($self) {
+    my $r = $self->_run_cmd(['branch', '--show-current']);
+    die $self->_format_git_error($r, 'Error detecting current branch') unless $r->{status};
+    return Mojo::Util::trim($r->{stdout});
+}
+
+sub _ssh_git_cmd ($git_args) {
+    return ['env', 'GIT_SSH_COMMAND=ssh -oBatchMode=yes', 'git', @$git_args];
+}
+
+sub get_remote_default_branch ($self, $url) {
+    my $r = run_cmd_with_log_return_error(_ssh_git_cmd(['ls-remote', '--symref', $url, 'HEAD']));
+    die $self->_format_git_error($r, "Error detecting remote default branch name for '$url'")
+      unless $r->{status} && $r->{stdout} =~ /refs\/heads\/(\S+)\s+HEAD/;
+    return $1;
+}
+
+sub git_clone_url_to_path ($self, $url, $path) {
+    my $r = $self->_run_cmd(_ssh_git_cmd(['clone', $url, $path]));
```

Comment:
```suggestion
sub git_clone_url_to_path ($self, $url) {
    my $r = $self->_run_cmd(_ssh_git_cmd(['clone', $url, $self->dir]));
```

Reply:
We might even rename the method to just `git_clone_url`

Reply:
I can't reproduce that error. If I do my suggested change, and I replace `$path` in the message one line below with `$self->dir` as well, and in the calling code I replace
```diff
- $git->git_clone_url_to_path($url, $path);
+ $git->git_clone_url_to_path($url);
```
the test passes

Reply:
The only thing that might be something to improve: the `-C dir` is actually not necessary and confusing when we do the initial clone.
Also we don't need it when doing the `get_remote_default_branch`. So maybe the `_prepare_git_command` should not be done in those cases.